provider-example/examples/server.rs

```diff
@@ -0,0 +1,107 @@
+use pki_types::CertificateDer;
+use pki_types::{PrivateKeyDer, PrivatePkcs8KeyDer};
+use rustls::server::Acceptor;
+use rustls::ServerConfig;
+use rustls_provider_example::PROVIDER;
+use std::io::Write;
+use std::sync::Arc;
+
+struct TestPki {
+    server_cert_der: CertificateDer<'static>,
+    server_key_der: PrivateKeyDer<'static>,
+}
+
+impl TestPki {
+    fn new() -> Self {
+        let alg = &rcgen::PKCS_ECDSA_P256_SHA256;
+        let mut ca_params = rcgen::CertificateParams::new(Vec::new());
+        ca_params
+            .distinguished_name
+            .push(rcgen::DnType::OrganizationName, "Rustls Server Acceptor");
+        ca_params
+            .distinguished_name
+            .push(rcgen::DnType::CommonName, "Example CA");
+        ca_params.is_ca = rcgen::IsCa::Ca(rcgen::BasicConstraints::Unconstrained);
+        ca_params.key_usages = vec![
+            rcgen::KeyUsagePurpose::KeyCertSign,
+            rcgen::KeyUsagePurpose::DigitalSignature,
+            rcgen::KeyUsagePurpose::CrlSign,
```

Comment:
Harmless to leave this, but if we're going to duplicate the code from the `server_acceptor.rs` example we might as well trim it down to what this example needs:
```suggestion
```